src/passwordcheck.js

```diff
@@ -27,12 +27,26 @@ define([
       PASSWORD_TOO_SHORT: 'PASSWORD_TOO_SHORT'
     };
 
-    function isStrongEnough(password) {
-      // Check for passwords that at least contain a number and an alphabet,
-      // or if alphabets, then at least (minLength + 4) characters long
-      var regexString = '((?=.*[a-zA-Z])(?=.*[0-9])[A-Za-z0-9!@#$%^&*()_+ ]{' + minLength + ',' + (minLength + 4) + '})|([A-Za-z0-9!@#$%^&*()_+ ]{' + (minLength + 4) + ',})';
```

Comment:
```
IIUC, this is a dictionary of 8-12 character words that contain only a combination of two or more classes of: letters, numbers, and punctuation. Passwords > 12 characters we consider "safe" even if they are all digits, or all numbers.
```
Shane, yes your understanding is correct. It contains passwords of 8-12 characters, with at least one special character inside of them.
1. I will try and simplify the regex so that its clear what its doing.
2. Sounds good, I can do that :)